components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -362,11 +357,51 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   std::set<GroupRequest>   m_required_group_requests;
   std::set<GroupRequest>   m_computed_group_requests;
 
+  // List of property checks for fields
+  using prop_check_ptr = std::shared_ptr<FieldPropertyCheck<Real>>; 
+  std::list<std::pair<FieldIdentifier,prop_check_ptr>> m_property_checks_in;
+  std::list<std::pair<FieldIdentifier,prop_check_ptr>> m_property_checks_out;
+
   // This process's copy of the timestamp, which is set on initialization and
   // updated during stepping.
   TimeStamp m_time_stamp;
 };
 
+// ====================== IMPLEMENTATION ======================= //
+
+template<RequestType RT, template<typename> class FPC, typename... Args>
+void AtmosphereProcess::add_property_check (const FieldIdentifier& fid, const Args... args) {
```

Comment:
What do you think about making this function idempotent? In other words: if you try to add a property check that's already been added this way, the method does nothing. Might make things easier to analyze.

Reply:
I was indeed trying to work on a way to ensure that the same FPC (**for the same field**) is not added twice. It would require a couple more maps to be scanned, but I opted for the easy (lazy?) way, and didn't do it.

To be fair, though, now that the FPC's are added/run inside each Atm Proc separately, I find it harder to have duplicates. It would probably only happen in case of copy+paste, when a dev is trying to add a FPC for a bunch of fields, but forgets to change the name of the field in one of them.